pkg/sql/schemachanger/scplan/internal/opgen/opgen_column_not_null.go

```diff
@@ -28,7 +28,10 @@ func init() {
 				}),
 			),
 			to(scpb.Status_VALIDATED,
-				emit(func(this *scpb.ColumnNotNull) *scop.ValidateColumnNotNull {
+				emit(func(this *scpb.ColumnNotNull, md *opGenContext) *scop.ValidateColumnNotNull {
+					if checkIfDescriptorIsAdded(this.TableID, md) {
+						return nil
+					}
```

Comment:
I don't think that this works. Assuming that CREATE TABLE was supported, consider
```
BEGIN;
CREATE TABLE t (x INT PRIMARY KEY);
INSERT INTO t VALUES (1);
ALTER TABLE t ADD COLUMN y INT NOT NULL DEFAULT 42;
COMMIT;
```
We still need to backfill and validate in that case, despite the relation descriptor being added.

What we really want is backfills and validation being conditional on the descriptor being added _and_ the relation being non-empty. We do have TableData and IndexData elements to model the presence of data. Right now these are used for DROP and we just assume that the data always exists in any case, but it needn't be so.

Anyway that's out of scope but `checkIfDescriptorIsAdded` should be expanded to also check for the presence of a ToPublic `TableData` or `IndexData` target, and those targets need to be set by the builder as well.

Reply:
Good point. I added a check for this but, for the longer term we need to figure out how to decompose emptiness in terms of round trips and everything (feels like something in collections)